Add React Hooks for customization (part 14) (#2652)
microsoft/BotFramework-WebChat: samples/10.b.customization-password-input/README.md

We'll start by using the [host with React sample](../03.a.host-with-react) as our Web Chat React template.

> In this sample, we will build a new React component that will be show when we receive an event activity from the bot named `passwordInput`. It will show a password box. When the user press <kbd>ENTER</kbd> to submit the box, we will send a post back activity back to the bot.

Reviewer: ```suggestion
In this sample, we will build a new React component that will be shown when we receive an event activity from the bot named `passwordInput`. It will show a password box. When the user presses <kbd>ENTER</kbd> to submit the box, a post back activity will be sent back to the bot.
```

Let's start by building the React component called `PasswordInputActivity`.
This component will render a simple form that asks the user to input their Two-Factor Authentication code.

```jsx
const PasswordInputActivity = () => {
   return (
      <form>

>          <labe >

Reviewer: ```suggestion
         <label>
```

            <span>Please input your 2FA code</span>
            <input
               autoFocus={true}
               type="password"
            />
         </label>
      </form>
   );
};
}
```

> Next, build our CSS and apply class names to our component.

Reviewer: ```suggestion
Next, build the CSS and apply class names to the component.
```

```css
.passwordInput {
   background-color: Red;
   border-radius: 3px;
   color: White;
   display: flex;
   font-family: Calibri, 'Helvetica Neue', Arial, sans-serif;
   margin: 10px;
   padding: 5px;
}

.passwordInput .passwordInput__box {
   display: flex;
   flex: 1;
}

.passwordInput .passwordInput__label {
   padding: 10px;
}

.passwordInput .passwordInput__input {
   border: 0px;
   border-radius: 3px;
   flex: 1;
   letter-spacing: .5em;
   outline: 0;
   padding: 0 10px;
   width: 100%;
}

.passwordInput .passwordInput__input:disabled {
   background-color: rgba(255, 255, 255, .5);
   color: White;
}
```

Then, apply the style sheet to our React component.

```diff
   return (
-     <form>
+     <form className="passwordInput">
-        <label>
+        <label className="passwordInput__box">
-           <span>Please input your 2FA code</span>
+           <span className="passwordInput__label">Please input your 2FA code</span>
            <input
               autoFocus={true}
+              className="passwordInput__input"
               type="password"
            />
         </label>
      </form>
   );
```

> Then, add business logic to the component:

Reviewer: ```suggestion
Next, add the business logic to the component:
```

> - When the user type on the password box, save the code.

Reviewer: ```suggestion
- When the user types in the password box, save the code.
```
Code? Do you mean text?

> - When the user press <kbd>ENTER</kbd> on the password box, submit the code and disable the password box.

Reviewer: ```suggestion
- When the user presses <kbd>ENTER</kbd> inside the password box, submit the code and disable the password box.
```
submit the input text?

> This component will have a state object with `code` and `submitted` values, initially set to an empty string and `false`, respectively. `code` will store the  code, and `submitted` will store a boolean value to indicate whether the code has been submitted or not. If the code has been submitted, disable the input box to prevent resubmission.

Reviewer: `code` should be renamed to `twoFACode` or something. Just `code` is confusing.

> This component will have two callback functions, called `handleCodeChange` and `handleSubmit`. The former will be called when the code has changed, it will save the value into `code` state. The latter will be called when the form is submitted (user press <kbd>ENTER</kbd> on the password box). It will submit the code via `sendPostBack`, and disable itself to prevent resubmission by updating `submitted` to `true`.

Reviewer: ```suggestion
This component will have two callback functions, called `handleCodeChange` and `handleSubmit`. The former will be called when the code has changed, and it will save the value into the `code` state. The latter will be called when the form is submitted (user press <kbd>ENTER</kbd> on the password box). It will submit the code via `sendPostBack`, and disable itself to prevent resubmission by updating `submitted` to `true`.
```

> The `sendPostBack` function will be retrieve from Web Chat hooks via `useSendPostBack` function.

Reviewer: ```suggestion
The `sendPostBack` function will be retrieved from Web Chat hooks via `useSendPostBack` function.
```
